packages-exp/app-check-exp/src/internal-api.ts

```diff
@@ -271,11 +279,22 @@ function notifyTokenListeners(
   app: FirebaseApp,
   token: AppCheckTokenResult
 ): void {
-  const listeners = getState(app).tokenListeners;
+  const observers = getState(app).tokenObservers;
 
-  for (const listener of listeners) {
+  for (const observer of observers) {
     try {
-      listener(token);
+      if (observer.error) {
+        // If this listener has an error handler, handle errors differently
+        // from successes.
+        if (token.error) {
+          observer.error(token.error);
+        } else {
+          observer.next(token);
+        }
+      } else {
+        // Otherwise return the token, whether or not it has an error field.
+        observer.next(token);
+      }
```

Comment:
It is not correct for 3p listeners - If there is no error handler, the error should just be ignored. 

Do we handle both 2p and 3p listeners in this function? If so, they need to be handled differently, where 2p listeners always get the token regardless of the error state, and 3p listeners get a real token in the success handler, and errors in error handler and errors should be ignored if there is no error handler.

That also means we need a way to tell 2p and 3p listeners apart.

Reply:
I probably missed it in the PR for v8, but we need to do the same in v8 too.

Reply:
I added a `type` property to the observers. Let me know if it works, and if the names are ok.